OnProfileUpdate/handler.ts

```diff
@@ -0,0 +1,200 @@
+                      getPreviousValidatedEmail(fiscalCode, newVersion)
+                    )
+                )
+              )
+      )
+    )
+  );
+
+const deleteProfileEmail = (
+  profileEmail: ProfileEmail
+): RTE.ReaderTaskEither<
+  DataTableProfileEmailsRepository,
+  Error,
+  void
+> => dataTableProfileEmailsRepository =>
+  TE.tryCatch(
+    () => dataTableProfileEmailsRepository.delete(profileEmail),
+    error =>
+      error instanceof Error
+        ? error
+        : new Error("error deleting ProfileEmail from table storage")
+  );
+
+const insertProfileEmail = (
+  profileEmail: ProfileEmail
+): RTE.ReaderTaskEither<
+  DataTableProfileEmailsRepository,
+  Error,
+  void
+> => dataTableProfileEmailsRepository =>
+  TE.tryCatch(
+    () => dataTableProfileEmailsRepository.insert(profileEmail),
+    error =>
+      error instanceof Error
+        ? error
+        : new Error("error inserting ProfileEmail from table storage")
+  );
+
+const upsertProfileEmail = ({
+  email,
+  fiscalCode,
+  version
+}: Omit<ProfileDocument, "isEmailValidated">) => ({
+  dataTableProfileEmailsRepository,
+  profileModel
+}: Omit<IDependencies, "logger">) =>
+  pipe(
+    version - 1,
+    NonNegativeInteger.decode,
+    E.fold(
+      () => TE.right<Error, void>(void 0),
+      previousVersion =>
+        pipe(
+          profileModel,
+          getPreviousValidatedEmail(fiscalCode, previousVersion),
+          TE.chainW(
+            flow(
+              //
+              O.foldW(
+                () =>
+                  pipe(
+                    dataTableProfileEmailsRepository,
+                    insertProfileEmail({ email, fiscalCode })
+                  ),
+                previousEmail =>
+                  pipe(
+                    email === previousEmail
+                      ? TE.right<Error, void>(void 0)
+                      : pipe(
+                          pipe(
+                            dataTableProfileEmailsRepository,
+                            deleteProfileEmail({
+                              email: previousEmail,
+                              fiscalCode
+                            })
+                          ),
+                          TE.chain(() =>
+                            pipe(
+                              dataTableProfileEmailsRepository,
+                              insertProfileEmail({ email, fiscalCode })
+                            )
+                          )
+                        )
+                  )
+              )
+            )
+          )
+        )
+    )
+  );
+
+const handleDocument = (
+  document: unknown
+): RTE.ReaderTaskEither<IDependencies, Error, void> => ({
+  dataTableProfileEmailsRepository,
+  profileModel,
+  logger
+}) =>
+  pipe(
+    document,
+    ProfileDocument.decode,
+    E.fold(
+      () => TE.right(void 0),
+      ({ email, fiscalCode, version }) =>
+        version === 0
+          ? pipe(
+              dataTableProfileEmailsRepository,
```

Comment:
if you work with `RTE<IDeps, ...>` instead, yor pipes would be simpler, as you don't have to pass down deps.